### PR TITLE
Add support for Cloud Deep Scan

### PR DESCRIPTION
**Pull request by s-kovacevic**

### CODE REVIEW

File: ReversingLabs/SDK/clouddeepscan.py [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+        :param sample_hash: sha1 hash of sample content
+        :type sample_hash: str
+        :param report_output_path: relative or absolute path where output will be saved with file name included
+        e.g. "reports/report1.json"
+        :type report_output_path: str
+        :raises CloudDeepScanException: if report download fails in any way
+        """
+        response = self.__api_request(
+            method="GET",
+            endpoint="/api/v1/reports",
```
Comment (MislavReversingLabs) [inline ReversingLabs/SDK/clouddeepscan.py:91]: Maybe pull this endpoint out as a variable? Same goes for the params arg. That can enable easier changes in the future.

File: ReversingLabs/SDK/clouddeepscan.py [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+        :type file_name: str
+        :param file_size: size of the uploaded sample in bytes
+        :type file_size: int
+        :raises CloudDeepScanException: if API fails to create upload
+        :return: created upload details
+        :rtype: dict[any]
+        """
+        response = self.__api_request(
+            method="POST",
+            endpoint="/api/v1/uploads",
```
Comment (MislavReversingLabs) [inline ReversingLabs/SDK/clouddeepscan.py:122]: Basically same comment for pulling out the endpoint and params as variables for easier editing.

File: ReversingLabs/SDK/helper.py [hunk 3]
```diff
@@ -98,6 +98,10 @@ def __init__(self, message="This content is not acceptable"):
         super(NotAcceptableError, self).__init__(message)
 
 
+class CloudDeepScanException(Exception):
```
[the remark this answers is not in the document]
Comment (s-kovacevic) [reply ReversingLabs/SDK/helper.py:101]: Yes, I felt like this was sufficient for now. I didn't want to complicate things for the end user, if you feel like I should make more specific errors and error hierarchy let me know.

File: ReversingLabs/SDK/clouddeepscan.py (hunk 1 above)
Comment (s-kovacevic) [reply ReversingLabs/SDK/clouddeepscan.py]: 👍

File: ReversingLabs/SDK/clouddeepscan.py (hunk 2 above)
Comment (s-kovacevic) [reply ReversingLabs/SDK/clouddeepscan.py]: 👍

File: README.md [hunk 4]
```diff
@@ -501,3 +542,35 @@ results = titanium_scale.upload_sample_and_get_results(
...
+
+cloud_deep_scan = CloudDeepScan(
+    token_endpoint="https://exampletokenendpoint.reversinglabs.com/oauth2/token",
+    rest_hostname="https://example.clouddeepscan.com",
+    client_id="035e633b65d94",
+    client_secret="f2207d99a74fbe169e3eba035e633b6"
+)
+try:
+    submission_id = cloud_deep_scan.upload_sample(sample_path="/path/to/file/suspicious_file.exe")
+except CloudDeepScanException as e:
```
[the remark this answers is not in the document]
Comment (s-kovacevic) [reply README.md:561]: Just wanted to show in the example that it raises an exception I do not suggest users to handle their exceptions in production like that. 😄
I'm OK with removing it if you'd like.

**Review by MislavReversingLabs — APPROVED** [review]

**Review by MislavReversingLabs — APPROVED** [review]

**Pull request merged**